zml/buffer.zig

```diff
@@ -122,20 +147,40 @@ pub const Buffer = struct {
     /// return a Buffer with the given dimensions.
     pub fn fromSlice(platform: Platform, dimz: anytype, s: anytype) !Buffer {
         const sh = Shape.init(dimz, DataType.fromSliceElementType(s));
-        return from(platform, HostBuffer.fromBytes(sh, std.mem.sliceAsBytes(s)));
+        return from(platform, HostBuffer.fromBytes(sh, std.mem.sliceAsBytes(s)), .{});
     }
 
     /// Copies the given Zig slice to the accelerator memory and
     /// return a Buffer with the given dimensions.
     pub fn fromBytes(platform: Platform, sh: Shape, data: []const u8) !Buffer {
-        return from(platform, HostBuffer.fromBytes(sh, data));
+        return from(platform, HostBuffer.fromBytes(sh, data), .{});
     }
 
     /// Copies the given Zig array to the accelerator memory and
     /// return a Buffer using the array shape.
     pub fn fromArray(platform: Platform, arr: anytype) !Buffer {
         const host_buffer = HostBuffer.fromArray(&arr);
-        return try from(platform, host_buffer);
+        return try from(platform, host_buffer, .{});
```

Comment:
fromArray should wait by default, because the arrays are typically on the stack.